ThirdSemester/ParallelMatrixMultiplication/ParallelMatrixMultiplication/Matrix.cs

```diff
@@ -0,0 +1,93 @@
+﻿// <copyright file="Matrix.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+/// <summary>
+/// Provides utility functions for reading, writing, and generating matrices.
+/// </summary>
+public static class Matrix
+{
+    /// <summary>
+    /// Reads a matrix from a file.
+    /// </summary>
+    /// <param name="fileName">The path to the file containing the matrix.</param>
+    /// <returns>A matrix represented as a 2D array.</returns>
+    public static int[,] ReadMatrixFromFile(string fileName)
+    {
+        string[] lines = File.ReadAllLines(fileName);
+
+        if (lines.Length == 0)
+        {
+            throw new ArgumentException("Файл пуст или содержит некорректные данные.");
+        }
+
+        int rows = lines.Length;
+        int cols = lines[0].Split(' ').Length;
+        int[,] matrix = new int[rows, cols];
```

Comment:
```suggestion
        var matrix = new int[rows, cols];
```
Не пишите в объявлении один тип дважды